### PR TITLE
[WIP] Non-linear surgery

### PR DESCRIPTION
**Pull request by Crazylemon64**
If you remember the days of bay surgery (*shudder*), you'll remember that you were able to do steps freely, as long as you did the pre-requisites. This brings that system back in a far more maintainable manner, allowing you to do more than one thing while you've got someone's ribcage open.

:cl:Crazylemon
rscadd: Surgery is now non-linear
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Color me highly skeptical

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -124,9 +126,10 @@
 /mob/living/carbon/gib()
 	for(var/obj/item/organ/internal/I in internal_organs)
 		if(isturf(loc))
-			I.remove(src)
-			I.forceMove(get_turf(src))
-			I.throw_at(get_edge_target_turf(src,pick(alldirs)),rand(1,3),5)
+			var/obj/item/thing = I.remove(src)
+			thing.forceMove(get_turf(src))
+			spawn()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/carbon.dm]: don't spawn.

**Comment by Landerlow** [comment]
At the moment it is already possible to repair bones via organ manipulation. So what else would be possible?

**Comment by Crazylemon64** [comment]
@Landerlow 
A very small set of examples:
* Opening up someone's body, doing a cavity implant, then fixing their damaged internal organs
* Dethralling a patient, then giving them a neural augment
* Remove bullet shrapnel, as well as patch internal bleeding, without having to close and re-select in between

Most prompts are also removed - instead you'll do the operation that corresponds to your tool. Only if something ambiguous comes up, do you get a prompt to choose between different surgery steps.

**Comment by FlattestGuitar** [comment]
Sweet Jesus, yes.

If it works without the massive amount of bugs we had the last time around, that's a  :+1: x :100:

**Comment by ktoma36** [comment]
How would this function with ghetto surgery? As I understand multiple tools can be used for different parts of surgeries

**Comment by FlattestGuitar** [comment]
You still have to do X thing in a given moment. If you're holding the right tool, that surgery will be attempted. Not much magic to it.

File: code/modules/mob/living/carbon/brain/posibrain.dm [hunk 2]
```diff
@@ -71,21 +71,29 @@
 	return
 
 /obj/item/device/mmi/posibrain/transfer_identity(var/mob/living/carbon/H)
-	name = "positronic brain ([H])"
+	name = "Positronic brain ([H])"
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/brain/posibrain.dm]: capitalisation power creep 🤣

File: code/modules/surgery/bones.dm [hunk 3]
```diff
@@ -53,17 +53,17 @@
 	..()
 
 /datum/surgery_step/glue_bone/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		var/obj/item/organ/external/affected = target.get_organ(target_zone)
-		user.visible_message("<span class='notice'> [user] applies some [tool] to [target]'s bone in [affected.name]</span>", \
-			"<span class='notice'> You apply some [tool] to [target]'s bone in [affected.name] with \the [tool].</span>")
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	user.visible_message("<span class='notice'>[user] applies some [tool] to [target]'s bone in [affected]</span>", \
```
Comment (SamHPurp) [inline code/modules/surgery/bones.dm]: full stop at end

File: code/modules/surgery/bones.dm [hunk 4]
```diff
@@ -30,7 +30,7 @@
 
 //surgery steps
 /datum/surgery_step/glue_bone
-	name = "mend bone"
+	name = "Mend bone"
```
Comment (SamHPurp) [inline code/modules/surgery/bones.dm]: You seem to have done this inconsistently with the other steps, `set_bone` for example remains uncapitalised.

File: code/modules/surgery/organs_internal.dm [hunk 5]
```diff
@@ -1,432 +1,527 @@
...
+	return 1
 
-	if(is_int_organ(tool))
-		current_type = "insert"
-		I = tool
-		if(target_zone != I.parent_organ || target.get_organ_slot(I.slot))
-			to_chat(user, "<span class='notice'>There is no room for [I] in [target]'s [parse_zone(target_zone)]!</span>")
-			return -1
+/datum/surgery_step/internal/manipulate_organs/repair/fail_step(mob/living/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/active_surgery/surgery)
+	user.visible_message("<span class='warning'>[user]'s hand slips, getting mess and tearing the inside of [target]'s [affected.name] with \the [tool]!</span>", \
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `getting mess`?

File: code/modules/surgery/organs_internal.dm [hunk 6]
```diff
@@ -1,432 +1,527 @@
...
+	return 0
+
+/datum/surgery_step/internal/manipulate_organs/repair/robo
+	name = "Manipulate organs (repair (robotic))"
+	allowed_tools = list(/obj/item/stack/nanopaste = 100, /obj/item/weapon/screwdriver = 70, /obj/item/weapon/bonegel = 30)
+	steps_this_can_pop = list(/datum/surgery_step/internal/manipulate_organs/repair/robo)
+
+/datum/surgery_step/internal/manipulate_organs/repair/robo/can_use(mob/living/user, mob/living/carbon/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	if(!hasorgans(target))
+		user << "They do not have organs to mend!"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: reee

File: code/modules/surgery/organs_internal.dm [hunk 7]
```diff
@@ -1,432 +1,527 @@
...
+		return 0
+	if(!ishuman(user))
+		return 0
+	var/mob/living/carbon/human/H = target
+	var/obj/item/organ/external/affected = H.get_organ(target_zone)
+	if(!affected)
+		user << "<span class='warning'>They do not have a [target_zone]!</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 8]
```diff
@@ -1,432 +1,527 @@
...
+		return
+	for(var/obj/item/organ/internal/I in affected.internal_organs)
+		if(I && I.damage > 0 && I.robotic == 2)
+			return 1
+	user << "There are no damaged organs that you can repair with this!"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 9]
```diff
@@ -1,432 +1,527 @@
...
+		if(I && I.robotic == 2)
+			I.surgeryize()
+			if(I.damage > 0)
+				user.visible_message("<span class='notice'>[user] repairs [target]'s [I.name] with [tool_name].</span>", \
+				"<span class='notice'>You repair [target]'s [I.name] with [tool_name].</span>" )
+				I.damage = 0
+	return 1
+
+/datum/surgery_step/internal/manipulate_organs/repair/robo/fail_step(mob/living/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/active_surgery/surgery)
+	user.visible_message("<span class='warning'>[user]'s hand slips, getting mess and tearing the inside of [target]'s [affected.name] with \the [tool]!</span>", \
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `getting mess`

File: code/modules/surgery/organs_internal.dm [hunk 10]
```diff
@@ -1,432 +1,527 @@
...
+	"<span class='warning'>Your hand slips, getting mess and tearing the inside of [target]'s [affected.name] with \the [tool]!</span>")
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `getting mess`

File: code/modules/surgery/organs_internal.dm [hunk 11]
```diff
@@ -1,432 +1,527 @@
...
+	if(ishuman(target))
+		H = target
+		affected = H.get_organ(user.zone_sel.selecting)
+
+	var/removable_count = 0
+	for(var/obj/item/organ/internal/O in organs)
+		if(O.robotic != 2)
+			removable_count++
+	if(removable_count == 0)
+		user << "<span class='notice'>There are no removeable organs in [target]'s [parse_zone(target_zone)]!</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

also `removeable` -> `removable`

File: code/modules/surgery/organs_internal.dm [hunk 12]
```diff
@@ -1,432 +1,527 @@
...
+	var/mob/living/carbon/human/H
+	if(ishuman(target))
+		H = target
+		affected = H.get_organ(target_zone)
+	var/removable_count = 0
+	for(var/obj/item/organ/internal/O in organs)
+		if(O.robotic == 2)
+			removable_count++
+	if(removable_count == 0)
+		user << "<span class='notice'>There are no removeable organs in [target]'s [parse_zone(target_zone)]!</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

also `removeable` -> `removable`

File: code/modules/surgery/organs_internal.dm [hunk 13]
```diff
@@ -1,432 +1,527 @@
...
 
-		var/dam_amt = 2
+	if(!..())
+		return 0
 
-		if(istype(tool, /obj/item/stack/medical/bruise_pack/advanced))
-			target.adjustToxLoss(5)
+	if(istype(tool, /obj/item/weapon/reagent_containers/food/snacks/organ))
+		// I mean, you probably won't be able to reach this code anyways
+		user << "<span class='warning'>[tool] was bitten by someone! It's too damaged to use!</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 14]
```diff
@@ -1,432 +1,527 @@
...
+		return 0
+	if(tool.flags & NODROP)
+		user << "<span class='warning'>[tool] is stuck to you! You can't drop it!</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 15]
```diff
@@ -1,432 +1,527 @@
...
+		return 0
 
-		else if(istype(tool, /obj/item/stack/medical/bruise_pack) || istype(tool, /obj/item/stack/nanopaste))
-			dam_amt = 5
-			target.adjustToxLoss(10)
-			affected.createwound(CUT, 5)
+	if(target_zone != I.parent_organ || target.get_organ_slot(I.slot))
+		user << "<span class='notice'>There is no room for [I] in [target]'s [parse_zone(target_zone)]!</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 16]
```diff
@@ -1,432 +1,527 @@
...
+		return 0
 
-		for(var/obj/item/organ/internal/I in affected.internal_organs)
-			if(I && I.damage > 0 && !(I.tough))
-				I.take_damage(dam_amt,0)
+	if(I.damage > (I.max_damage * 0.75))
+		user << "<span class='notice'>\The [I] is in no state to be transplanted.</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 17]
```diff
@@ -1,432 +1,527 @@
...
+		return 0
 
+	if(target.get_int_organ(I))
+		user << "<span class='warning'>\The [target] already has [I].</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 18]
```diff
@@ -1,432 +1,527 @@
...
+		user.visible_message("[user] starts transplanting \the [tool] into [target]'s [parse_zone(target_zone)].", \
+		"You start transplanting \the [tool] into [target]'s [parse_zone(target_zone)].")
 
+
+/datum/surgery_step/internal/manipulate_organs/insert/end_step(mob/living/user, mob/living/carbon/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	user.drop_item()
+	I.insert(target)
+	spread_germs_to_organ(I, user)
+	if(!user.canUnEquip(I, 0))
+		user << "<span class='warning'>[I] is stuck to your hand, you can't put it in [target]!</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 19]
```diff
@@ -1,432 +1,527 @@
...
-				"<span class='warning'> Your hand slips, damaging [target]'s [affected.name] with [tool]!</span>")
-				affected.createwound(BRUISE, 20)
-			else
-				user.visible_message("<span class='warning'> [user]'s hand slips, damaging [target]'s [parse_zone(target_zone)] with [tool]!</span>", \
-				"<span class='warning'> Your hand slips, damaging [target]'s [parse_zone(target_zone)] with [tool]!</span>")
-		else
-			user.visible_message("[user] can't seem to extract anything from [target]'s [parse_zone(target_zone)]!",
-				"<span class='notice'>You can't extract anything from [target]'s [parse_zone(target_zone)]!</span>")
+	if(!M.brainmob || !M.brainmob.client || !M.brainmob.ckey || M.brainmob.stat >= DEAD)
+		user << "<span class='danger'>That brain is not usable.</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 20]
```diff
@@ -1,432 +1,527 @@
...
-			user.visible_message(msg, self_msg)
-			affected.fracture()
-		else
-			var/msg = "<span class='warning'> [user]'s hand slips, tearing the skin!</span>"
-			var/self_msg = "<span class='warning'> Your hand slips, tearing skin!</span>"
-			user.visible_message(msg, self_msg)
-		if(affected)
-			affected.createwound(BRUISE, 20)
+	if(!(affected && affected.status & ORGAN_ROBOT))
+		user << "<span class='danger'>You cannot install a computer brain into a meat enclosure.</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 21]
```diff
@@ -1,432 +1,527 @@
...
 		return 0
 
+	if(!H.species.has_organ["brain"])
+		user << "<span class='danger'>You're pretty sure [H.species.name_plural] don't normally have a brain.</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

`don't` -> `doesn't` - grammatically, maybe? Can someone else confirm? My brain is feeling a bit fried.

File: code/modules/surgery/organs_internal.dm [hunk 22]
```diff
@@ -1,432 +1,527 @@
...
+		return 0
+
+	if(H.get_int_organ(/obj/item/organ/internal/brain))
+		user << "<span class='danger'>Your subject already has a brain.</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 23]
```diff
@@ -1,432 +1,527 @@
...
+		return 0
+
+	if(M.flags & NODROP)
+		user << "<span class='warning'>\The [M] is stuck to you, you can't drop it!</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 24]
```diff
@@ -1,432 +1,527 @@
...
+
+/datum/surgery_step/internal/manipulate_organs/insert/mmi/begin_step(mob/living/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/active_surgery/surgery)
+	user.visible_message("[user] starts installing \the [tool] into [target]'s [affected.name].", \
+	"You start installing \the [tool] into [target]'s [affected.name].")
 
-	return 0
 
+/datum/surgery_step/internal/manipulate_organs/insert/mmi/end_step(mob/living/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/active_surgery/surgery)
+	if(!user.canUnEquip(tool, 0))
+		user << "<span class='warning'>[tool] is stuck to your hand, you can't put it in [target]!</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: `user <<`

File: code/modules/surgery/organs_internal.dm [hunk 25]
```diff
@@ -519,25 +614,25 @@
 	..()
 
 /datum/surgery_step/retract_carapace/end_step(mob/living/user, mob/living/carbon/target, target_zone, obj/item/tool,datum/surgery/surgery)
-	var/msg = "<span class='notice'> [user] keeps the incision open on [target]'s [target_zone] with [tool]</span>."
-	var/self_msg = "<span class='notice'> You keep the incision open on [target]'s [target_zone] with [tool].</span>"
+	var/msg = "<span class='notice'>[user] keeps the incision open on [target]'s [target_zone] with [tool]</span>."
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: Missing period/fullstop

File: code/modules/surgery/organs_internal.dm [hunk 26]
```diff
@@ -519,25 +614,25 @@
...
+	var/self_msg = "<span class='warning'>Your hand slips, tearing the edges of incision on [target]'s [target_zone] with [tool]!</span>"
 	if(target_zone == "chest")
-		msg = "<span class='warning'> [user]'s hand slips, damaging several organs [target]'s torso with [tool]!</span>"
-		self_msg = "<span class='warning'> Your hand slips, damaging several organs [target]'s torso with [tool]!</span>"
+		msg = "<span class='warning'>[user]'s hand slips, damaging several organs [target]'s torso with [tool]!</span>"
+		self_msg = "<span class='warning'>Your hand slips, damaging several organs [target]'s torso with [tool]!</span>"
 	if(target_zone == "groin")
-		msg = "<span class='warning'> [user]'s hand slips, damaging several organs [target]'s lower abdomen with [tool]</span>"
-		self_msg = "<span class='warning'> Your hand slips, damaging several organs [target]'s lower abdomen with [tool]!</span>"
+		msg = "<span class='warning'>[user]'s hand slips, damaging several organs [target]'s lower abdomen with [tool]</span>"
```
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: Missing period/fullstop

File: code/modules/surgery/other.dm [hunk 27]
```diff
@@ -251,13 +252,18 @@
 //					Dethrall Shadowling 						//
 //////////////////////////////////////////////////////////////////
 /datum/surgery/remove_thrall
-	name = "Remove Shadow Tumor"
-	steps = list(/datum/surgery_step/generic/cut_open, /datum/surgery_step/generic/clamp_bleeders, /datum/surgery_step/generic/retract_skin, /datum/surgery_step/open_encased/saw,/datum/surgery_step/open_encased/retract, /datum/surgery_step/internal/dethrall, /datum/surgery_step/glue_bone, /datum/surgery_step/set_bone,/datum/surgery_step/finish_bone,/datum/surgery_step/generic/cauterize)
-	possible_locs = list("head", "chest", "groin")
+	name = "Cleanse contaminations (hard shell)"//RENAME MEH
+	steps = list(/datum/surgery_step/generic/cut_open, /datum/surgery_step/generic/clamp_bleeders, /datum/surgery_step/generic/retract_skin, /datum/surgery_step/open_encased/saw,/datum/surgery_step/open_encased/retract, /datum/surgery_step/dethrall)
+	possible_locs = list("head", "chest", "groin") // Reasonable places to find a central nervous system
```
Comment (SamHPurp) [inline code/modules/surgery/other.dm]: Can the tumor go anywhere but the `head` even?

File: code/modules/surgery/other.dm [hunk 28]
```diff
@@ -331,15 +363,24 @@
...
+	var/obj/item/organ/tumor = new /obj/item/organ/internal/shadowtumor(get_turf(target))
+	tumor.set_dna(target.dna)
+	user.put_in_hands(tumor)
 	return 1
+
+/datum/surgery_step/internal/dethrall/fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	var/braincase = target.named_organ_parent("brain")
+	if(prob(50))
+		user.visible_message("<span class='warning'>[user] slips and rips the tumor out from [target]'s [braincase]!</span>", \
+							 "<span class='warning'><b>You fumble and tear out [target]'s tumor!</span>")
```
Comment (SamHPurp) [inline code/modules/surgery/other.dm]: you don't close your `<b>` tag

File: code/modules/surgery/robotics.dm [hunk 29]
```diff
@@ -150,387 +181,135 @@
...
-		return 1
-	return 0
-
-/datum/surgery_step/robotics/manipulate_robotic_organs/fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
-
-	if(current_type == "mend")
-		if(!hasorgans(target))
-			return
-		var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	user.visible_message("<span class='notice'> [user] finishes patching damage to [target]'s [affected.name] with \the [tool].</span>", \
```
Comment (SamHPurp) [inline code/modules/surgery/robotics.dm]: superfluous space after the `<span>`

File: code/modules/surgery/robotics.dm [hunk 30]
```diff
@@ -150,387 +181,135 @@
...
+	"<span class='notice'> You finish patching damage to [target]'s [affected.name] with \the [tool].</span>")
```
Comment (SamHPurp) [inline code/modules/surgery/robotics.dm]: superfluous space after the `<span>`

File: code/modules/surgery/robotics.dm [hunk 31]
```diff
@@ -150,387 +181,135 @@
...
+	return 1
 
-		user.visible_message("<span class='warning'> [user]'s hand slips, gumming up the mechanisms inside of [target]'s [affected.name] with \the [tool]!</span>", \
-		"<span class='warning'> Your hand slips, gumming up the mechanisms inside of [target]'s [affected.name] with \the [tool]!</span>")
 
-		target.adjustToxLoss(5)
-		affected.createwound(CUT, 5)
+/datum/surgery_step/robotics/external/repair_brute/fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/active_surgery/surgery)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	user.visible_message("<span class='warning'> [user]'s [tool.name] slips, damaging the internal structure of [target]'s [affected.name].</span>",
```
Comment (SamHPurp) [inline code/modules/surgery/robotics.dm]: superfluous space after the `<span>`

File: code/modules/surgery/robotics.dm [hunk 32]
```diff
@@ -150,387 +181,135 @@
...
+	"<span class='warning'> Your [tool.name] slips, damaging the internal structure of [target]'s [affected.name].</span>")
```
Comment (SamHPurp) [inline code/modules/surgery/robotics.dm]: superfluous space after the `<span>`

File: code/modules/surgery/robotics.dm [hunk 33]
```diff
@@ -150,387 +181,135 @@
...
-		user.visible_message("<span class='warning'> [user]'s hand slips!</span>.", \
-		"<span class='warning'> Your hand slips!</span>")
-	else if(current_type == "finish")
-		var/obj/item/organ/external/affected = target.get_organ(target_zone)
-		user.visible_message("<span class='warning'> [user]'s [tool.name] slips, failing to close the hatch on [target]'s [affected.name].</span>",
-		"<span class='warning'> Your [tool.name] slips, failing to close the hatch on [target]'s [affected.name].</span>")
-	return 0
+/datum/surgery_step/robotics/external/repair_burn/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/active_surgery/surgery)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	user.visible_message("<span class='notice'> [user] finishes splicing cable into [target]'s [affected.name].</span>", \
```
Comment (SamHPurp) [inline code/modules/surgery/robotics.dm]: superfluous space after the `<span>`

File: code/modules/surgery/robotics.dm [hunk 34]
```diff
@@ -150,387 +181,135 @@
...
+	"<span class='notice'> You finishes splicing new cable into [target]'s [affected.name].</span>")
```
Comment (SamHPurp) [inline code/modules/surgery/robotics.dm]: superfluous space after the `<span>`

File: code/modules/surgery/robotics.dm [hunk 35]
```diff
@@ -150,387 +181,135 @@
...
+	affected.heal_damage(0,rand(30,50),1,1)
+	return 1
 
+/datum/surgery_step/robotics/external/repair/fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/active_surgery/surgery)
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	user.visible_message("<span class='warning'> [user] causes a short circuit in [target]'s [affected.name]!</span>",
```
Comment (SamHPurp) [inline code/modules/surgery/robotics.dm]: superfluous space after the `<span>`

File: code/modules/surgery/robotics.dm [hunk 36]
```diff
@@ -150,387 +181,135 @@
...
+	"<span class='warning'> You cause a short circuit in [target]'s [affected.name]!</span>")
```
Comment (SamHPurp) [inline code/modules/surgery/robotics.dm]: superfluous space after the `<span>`

File: code/modules/surgery/other.dm (hunk 27 above)
Comment (MarcellusPye) [inline code/modules/surgery/other.dm]: IPC tumors are in the chest I *think*.

**Comment by taukausanake** [comment]
Half of me wonders what @tigercat2000 would be going through had @SamHPurp not found all those spaces and capitals. The other half imagines a massive breakdown of Tiger's brain :laughing:

File: code/_onclick/item_attack.dm [hunk 37]
```diff
@@ -32,26 +32,8 @@
...
-				else
-					return 1
-
-		if(istype(src,/obj/item/weapon/screwdriver) && M.get_species() == "Machine")
-			if(!attempt_initiate_surgery(src, M, user))
-				return 0
-			else
-				return 1
-		if(is_sharp(src))
+	if (can_operate(M))  //Checks if mob is lying down on table for surgery
```
Comment (KasparoVy) [inline code/_onclick/item_attack.dm]: spess b/w `if` and `(can_operate)`

File: code/modules/surgery/organs_internal.dm (hunk 21 above)
Comment (EvadableMoxie) [inline code/modules/surgery/organs_internal.dm]: If [H.species.name_plural] generates the plural form of the name of the species, then 'don't is correct:

"You're pretty sure Humans don't normally have a brain."

"You're pretty sure Humans doesn't normally have a brain."

File: code/modules/surgery/other.dm (hunk 27 above)
Comment (EvadableMoxie) [inline code/modules/surgery/other.dm]: Tumors are always in the head for everyone but IPCs, who have it in their chest.

File: code/modules/surgery/organs_internal.dm (hunk 21 above)
Comment (SamHPurp) [inline code/modules/surgery/organs_internal.dm]: 👍 thanks @EvadableMoxie 

at least I'm not alone with that one tripping me up, thanks thumbs-uppers, we're all in this incorrectly together! ;)

File: code/modules/surgery/other.dm (hunk 27 above)
Comment (SamHPurp) [inline code/modules/surgery/other.dm]: Then I guess the query remains with the `groin` entry

File: code/modules/surgery/helpers.dm [hunk 38]
```diff
@@ -1,5 +1,6 @@
-/proc/attempt_initiate_surgery(obj/item/I, mob/living/M, mob/user, var/override )
+/proc/attempt_initiate_surgery(var/obj/item/I, var/mob/living/M, var/mob/user)
```
Comment (RemieRichards) [inline code/modules/surgery/helpers.dm]: >ADDING var/ to proc args

You HEATHEN.

File: code/modules/surgery/other.dm (hunk 27 above)
Comment (MarcellusPye) [inline code/modules/surgery/other.dm]: Lewd shadowlings? This should probably be changed, also IPC ❄️

File: code/modules/surgery/helpers.dm (hunk 38 above)
Comment (Crazylemon64) [inline code/modules/surgery/helpers.dm]: I brought this code back from like 1 and a half years ago, my apologies

**Comment by Crazylemon64** [comment]
excuse me while I burn out all the remnants of obsolete coding practices from this fossil

File: code/modules/surgery/other.dm (hunk 27 above)
Comment (Crazylemon64) [inline code/modules/surgery/other.dm]: It's not snowflake, this is generalized to work for central nervous system organs installed in any central part of the body.

**Comment by taukausanake** [comment]
> code back from like 1 and a half years ago

And now I'm trying to find the PR where this refactor could potentially be a revert too. Can't find anything though

**Comment by Crazylemon64** [comment]
This is not a reversion, I wrote this code a long time ago, a little after the organ refactor went through, but dropped it at the time

Now that some time has passed and a spot of time opened up I figured I could try and resuscitate this

**Comment by Crazylemon64** [comment]
begone thot

File: code/modules/surgery/bones.dm [hunk 39]
```diff
@@ -161,10 +163,10 @@
 	"You start to finish mending the damaged bones in [target]'s [affected.name] with \the [tool].")
 	..()
 
-/datum/surgery_step/finish_bone/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+/datum/surgery_step/finish_bone/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool, datum/active_surgery/surgery)
 	var/obj/item/organ/external/affected = target.get_organ(target_zone)
-	user.visible_message("<span class='notice'> [user] has mended the damaged bones in [target]'s [affected.name] with \the [tool].</span>"  , \
-		"<span class='notice'> You have mended the damaged bones in [target]'s [affected.name] with \the [tool].</span>" )
+	user.visible_message("<span class='notice'>[user] has mended the damaged bones in [target]'s [affected.name] with \the [tool].</span>"  , \
+		"<span class='notice'>You have mended the damaged bones in [target]'s [affected.name] with \the [tool].</span>" )
```
Comment (Crazylemon64) [inline code/modules/surgery/bones.dm]: My concern is that this might make it "Spacenerd's the right arm" or something

Comment (SamHPurp) [inline code/modules/surgery/bones.dm]: That makes sense.

**Comment by Fox-McCloud** [comment]
Darn it, travis, do your job.

**Comment by Crazylemon64** [comment]
@Fox-McCloud It's merge-conflicted, I'll take care of it

**Comment by alexkar598** [comment]
Added a commit 1 year ago?I'm confused.

**Comment by Puckaboo2** [comment]
SHUT UP AND TAKE MY MONEY!

It is so damn annoying to necessarily close up a limb after internal bleeding, just to open it up again to fix a broken bone.

Don't even get me started on when it is the chest or skull >.>

**Comment by ktoma36** [comment]
It would also allow for some faster fix ups when time is of the essence, the most prominent example would be a xenomorph infected individual and getting the dam embryo out in time. Especially since someone can pop withen 40 seconds of getting infected (yes this has happened to me and I did time it)

**Comment by HugoLuman** [comment]
Non-euclidean surgery when

**Comment by SamHPurp** [comment]
Misc note on this.... apparently clowns with clumsy can do surgery perfectly atm with no failures.... which seems goofy. Something to consider altering while you're refactoring surgery.

**Comment by taukausanake** [comment]
*casts Raise Dead*

Is this still being worked on? It sounds interesting but still kind of crazy that it could work

**Comment by Crazylemon64** [comment]
oh, I'm pretty sure it's "ready" in the state that it's in? Just, that it needs to be updated for any new surgeries that may have sprung up, and perhaps given a debug pass of testing on my end to verify I didn't introduce any quirks.

**Comment by GeneralChaos81** [comment]
Will this fix ghetto surgery? Decap'd heads can't be worked on at all with ghetto, and organ removal is finicky.

**Comment by Crazylemon64** [comment]
yeehaw 🤠

File: code/modules/mob/transform_procs.dm [hunk 40]
```diff
@@ -55,10 +55,8 @@
 	canmove = 0
 	icon = null
 	invisibility = 101
-	for(var/t in bodyparts)
-		qdel(t)
-	for(var/i in internal_organs)
-		qdel(i)
+	QDEL_NULL(bodyparts)
+	QDEL_NULL(internal_organs)
```
Comment (Fox-McCloud) [inline code/modules/mob/transform_procs.dm]: don't qdel lists; there's an other define for this.

**Comment by Crazylemon64** [comment]
(Don't merge this yet, I'm still ironing out a few kinks with specific surgeries, and the operating computer)

File: code/modules/mob/transform_procs.dm (hunk 40 above)
Comment (Crazylemon64) [reply code/modules/mob/transform_procs.dm]: oh yeah whoops

**Comment by Crazylemon64** [comment]
new_op_computer

File: code/modules/mob/living/carbon/brain/life.dm [hunk 41]
```diff
@@ -43,5 +43,23 @@
...
 		. = 1
 
+
 /mob/living/carbon/brain/breathe()
 	return
+
+/mob/living/carbon/brain/handle_vision()
+	..()
+
+	if (stat == DEAD || (XRAY in src.mutations))
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/brain/life.dm]: Spacing, src.

File: code/modules/mob/living/carbon/brain/posibrain.dm [hunk 42]
```diff
@@ -69,22 +69,29 @@
 	return
 
 /obj/item/mmi/posibrain/transfer_identity(var/mob/living/carbon/H)
-	name = "positronic brain ([H])"
+	name = "Positronic brain ([H])"
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/brain/posibrain.dm]: I feel like this should not be a proper noun, "He is holding Positronic brain (Bob)" vs "He is holding the positronic brain (Bob)"

File: code/modules/mob/living/carbon/carbon.dm (hunk 1 above)
Comment (tigercat2000) [reply code/modules/mob/living/carbon/carbon.dm]: ^

**Comment by Crazylemon64** [comment]
This PR's holding up the queue, so I'll close it until I get a stretch of time to de-conflict it. If someone else wants to take a stab at it, feel free

**Pull request closed without merging**